src/fabric/src/fabric_view_all_docs.erl

```diff
@@ -85,10 +85,13 @@ go(DbName, Options, QueryArgs, Callback, Acc0) ->
     Resp = case couch_util:get_value(namespace, Extra, <<"_all_docs">>) of
         <<"_local">> ->
             {ok, null};
-        _ ->
+        NS ->
```

Comment:
I don't have strong opinion on this. The thing I don't really like is the fact that we have multiple places where we check for different values for namespace. Instead of a plane `case` with clauses for every possible value we have a `NS` clause and then have a second conditional there.  Would it be an option to extract this into a fun and call it like so:
```
 Resp = case couch_util:get_value(namespace, Extra, <<"_all_docs">>) of
        <<"_all_docs">> ->
            get_doc_counts(DbName, fun fabric:get_doc_count/1);
        <<"_design">> ->
            get_doc_counts(DbName, fun fabric:get_design_doc_count/1);
       <<"_local">> ->
            {ok, null};
end
``` 

It is a pity we don't have helper function for it. Since we have this pattern everywhere. Essentially `get_doc_counts` is `spawn_and_wait`.

Reply:
Well, this is essentially just changing two-level `case` into one-level `case` and I agree this is indeed adds clarity, but as a trade off we are adding another level of abstraction with sending function as an argument (and those are usually annoying to debug) plus that awkward moment where we are sending `get_doc_count` function into `get_doc_counts` function, which I imagine would require some untangling on a casual code reading.

How about abstracting things on `fabric:get_doc_count` level with something like this?

```diff
diff --git a/src/fabric/src/fabric.erl b/src/fabric/src/fabric.erl
index 36ed67154..868a72c46 100644
--- a/src/fabric/src/fabric.erl
+++ b/src/fabric/src/fabric.erl
@@ -91,7 +91,13 @@ get_db_info(DbName) ->
     {error, atom()} |
     {error, atom(), any()}.
 get_doc_count(DbName) ->
-    fabric_db_doc_count:go(dbname(DbName)).
+    get_doc_count(DbName, <<"_all_docs">>).
+
+get_doc_count(DbName, <<"_all_docs">>) ->
+    fabric_db_doc_count:go(dbname(DbName));
+get_doc_count(DbName, <<"_design">>) ->
+    fabric_design_doc_count:go(dbname(DbName)).
+
 
 %% @equiv create_db(DbName, [])
 create_db(DbName) ->
diff --git a/src/fabric/src/fabric_view_all_docs.erl b/src/fabric/src/fabric_view_all_docs.erl
index ac16dac52..6a8b1546b 100644
--- a/src/fabric/src/fabric_view_all_docs.erl
+++ b/src/fabric/src/fabric_view_all_docs.erl
@@ -85,10 +85,10 @@ go(DbName, Options, QueryArgs, Callback, Acc0) ->
     Resp = case couch_util:get_value(namespace, Extra, <<"_all_docs">>) of
         <<"_local">> ->
             {ok, null};
-        _ ->
+        Namespace ->
             Timeout = fabric_util:all_docs_timeout(),
             {_, Ref0} = spawn_monitor(fun() ->
-                exit(fabric:get_doc_count(DbName))
+                exit(fabric:get_doc_count(DbName, Namespace))
             end),
             receive {'DOWN', Ref0, _, _, Result} ->
                 Result
```